src/Microsoft.Kusto.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -308,6 +320,13 @@ private async Task DoExecute(ReliableDataSourceConnection conn, CancellationToke
                 {
                     await ExecuteOnce(conn, cancellationToken);
                 }
+                catch (KustoRequestException exception)
+                {
+                    if (exception.FailureCode == 401) // Unauthorized
+                    {
+                        throw new KustoUnauthorizedException(exception.Message);
```

Comment:
Using custom exception to keep Kusto.Data references from bleeding up the stack.